src/backends/policy_engine/souffle/raksha_policy_datalog_emitter.cc

```diff
@@ -86,6 +87,7 @@ int main(int argc, char* argv[]) {
   // Invoke the parser for auth_logic and return datalog
   AuthLogicProgram program =
       raksha::ir::auth_logic::ParseProgram(policy_rules.value());
+  program = raksha::ir::auth_logic::InsertUniverseRelations(program);
```

Comment:
This whole PR is just about this line. The test for this PR is that there are no regressions of the existing tests. The followup work that will test this is to port the regression tests that the rust implementation had.